React hooks: best practices / fix all lint errors

Giraffe currently has 10 lint errors when checked by `eslint-plugin-react-hooks`

**EDIT**: this issue has been split into 2 separate issues. Only the lint errors remain in this issue. All lint warnings have been moved into a separate issue.

Two action items:
- [x] fix all lint errors (see list below)
- [x] activate the `eslint-plugin-react-hooks` rule by installing it

The lint errors are:

- [x] /giraffe/src/components/Geo.tsx
  46:3   error    React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  react-hooks/rules-of-hooks
  51:53  error    React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?   react-hooks/rules-of-hooks
  63:3   error    React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  react-hooks/rules-of-hooks

- [x] giraffe/src/components/MultiGrid.tsx
  601:28  error    React Hook "useForceUpdate" is called in function "forceUpdate" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter  react-hooks/rules-of-hooks

- [x] /giraffe/src/components/Plot.tsx
  16:19  error  React Hook "useRef" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks
  17:20  error  React Hook "useRef" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks


- [x] /giraffe/src/components/geo/CircleMarkerLayer.tsx
  83:9  error    React Hook "useLayoutEffect" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks

- [x] /giraffe/src/components/geo/PointMapLayer.tsx
  71:9  error    React Hook "useLayoutEffect" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks

- [x] /giraffe/src/utils/legend/useTooltipElement.ts
  33:5  error  React Hook "useAnnotationStyle" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks
  35:5  error  React Hook "useTooltipStyle" is called conditionally. React Hooks must be called in the exact same order in every component render     react-hooks/rules-of-hooks

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -17,6 +17,7 @@
     "plugin:react/recommended",
     "plugin:@typescript-eslint/recommended",
     "plugin:prettier/recommended",
+    "plugin:react-hooks/recommended",
     "prettier/react",
     "prettier/@typescript-eslint"
   ],
@@ -49,7 +50,8 @@
     "@typescript-eslint/no-var-requires": "off",
     "@typescript-eslint/prefer-interface": "off",
     "@typescript-eslint/camelcase": "off",
-    "@typescript-eslint/no-unused-vars": ["error",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
       {"varsIgnorePattern": "^_", "argsIgnorePattern": "^_"}
     ],
     "react/jsx-no-target-blank": "off",

--- a/giraffe/package.json
+++ b/giraffe/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@influxdata/giraffe",
-  "version": "2.17.0",
+  "version": "2.18.0",
   "main": "dist/index.js",
   "module": "src/index.js",
   "license": "MIT",
@@ -71,6 +71,7 @@
     "eslint-config-prettier": "^6.7.0",
     "eslint-plugin-prettier": "^3.1.1",
     "eslint-plugin-react": "^7.17.0",
+    "eslint-plugin-react-hooks": "^4.0.5",
     "file-loader": "^6.1.1",
     "he": "1.2.0",
     "intl-dateformat": "^0.1.1",

--- a/stories/package.json
+++ b/stories/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@influxdata/giraffe-stories",
-  "version": "2.17.0",
+  "version": "2.18.0",
   "license": "MIT",
   "scripts": {
     "lint": "eslint '{src,../giraffe/src}/**/*.{ts,tsx}'",

